Python/pystate.c

```diff
@@ -250,12 +255,36 @@ alloc_interpreter(void)
 static void
 free_interpreter(PyInterpreterState *interp)
 {
-    PyMem_RawFree(interp);
+    if (interp != &interp->runtime->_preallocated.interpreters_main) {
+        PyMem_RawFree(interp);
+    }
+}
+
+static void init_threadstate_static_data(PyThreadState *, PyThreadState *);
+
+static void
+init_interpreter_static_data(PyInterpreterState *interp,
+                             PyInterpreterState *other)
```

Comment:
Why does this take two thread states?

Reply:
"other" is the source for the statically-initialized data with which "interp" will be initialized.